### ISSUE TITLE
Class/method hierarchy actions support for clojure code

### ISSUE THREAD

**Issue by sdomin** [cursive-ide/cursive#1212]
### Intellij Java background
Putting the cursor on a java code line with a method or a class definition and invoking Navigate->Implementation(s) actions, brings a list of references to all places that implement/override the definition.
If one follows a reference, a dual action will become available: Navigate->Super Method or Navigate->Super Class. Invoking this action will open the original place in the editor.

There are also the "Type Hierarchy", "Method Hierarchy" and "Show UML Popup" actions that can show trees of supertypes and subtypes. (Try these on, e.g., clojure.lang.APersistentVector)

### Feature request
Seems like it would make sense to support all these actions for clojure's defprotocol and defmulti with corresponding extend-protocol and defmethods.
And also for deftype (where it should be possible to see references to clojure code when using "Implementation(s)" from java code).

E.g. it would be really nice to easily see the hierarchy for print-method/print-dup multimethods and to be able to navigate around it.

**Comment by cursive-ide** [comment]
Yes, this would definitely be great. I haven't investigated how to do this yet but it's definitely on my list. It's also required for some other operations such as renaming a protocol method and having that rename the implementations.
